testing/dynamodb/streaming/integration_test.go

```diff
@@ -0,0 +1,428 @@
+		AttrNumberSet: []float64{1.5, 2.5, 3.5},
+		AttrByteSet:   [][]byte{[]byte{48, 49}, []byte{50, 51}},
+		AttrStringSet: []string{"abc", "xyz"},
+		AttrList:      []interface{}{"str-1", 12.34, true},
+		AttrMap:       map[string]int{"key": 100},
+	}
+	av, err := dynamodbattribute.MarshalMap(dydbRecord)
+	if err != nil {
+		t.Fatalf("Got error marshalling new movie item: %s", err)
+	}
+
+	putItemInput := &dydb.PutItemInput{
+		Item:      av,
+		TableName: aws.String(tableName),
+	}
+
+	_, err = dydbClient.PutItem(putItemInput)
+	if err != nil {
+		t.Fatalf("Got error calling PutItem: %s", err)
+	}
+	log.Println("Successfully created table and put item for dynamodb")
+}
+
+func populateDynamoDBStreams(t *testing.T) {
+	tableName := "testtable"
+	dydbRecord := DydbRecord{
+		AttrString:    "efgh",
+		AttrInt:       20,
+		AttrFloat:     24.5,
+		AttrBool:      false,
+		AttrBytes:     []byte{48, 49},
+		AttrNumberSet: []float64{1.5, 2.5, 3.5},
+		AttrByteSet:   [][]byte{[]byte{48, 49}, []byte{50, 51}},
+		AttrStringSet: []string{"abc", "xyz"},
+		AttrList:      []interface{}{"str-1", 12.34, true},
+		AttrMap:       map[string]int{"key": 102},
+	}
+	av, err := dynamodbattribute.MarshalMap(dydbRecord)
+	if err != nil {
+		t.Fatalf("Got error marshalling new movie item: %s", err)
+	}
+
+	putItemInput := &dydb.PutItemInput{
+		Item:      av,
+		TableName: aws.String(tableName),
+	}
+
+	_, err = dydbClient.PutItem(putItemInput)
+	if err != nil {
+		t.Fatalf("Got error calling PutItem: %s", err)
+	}
+	log.Println("Successfully inserted item for streaming migration")
+}
+
+func RunStreamingMigration(t *testing.T, args string, projectID string, client *spanner.Client) error {
+	// Be aware that when testing with the command, the time `now` might be
+	// different between file prefixes and the contents in the files. This
+	// is because file prefixes use `now` from here (the test function) and
+	// the generated time in the files uses a `now` inside the command, which
+	// can be different.
+	cmd := exec.Command("bash", "-c", fmt.Sprintf("go run github.com/cloudspannerecosystem/harbourbridge %v", args))
+	var out, stderr bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("GCLOUD_PROJECT=%s", projectID),
+	)
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf(fmt.Sprintf("error while starting command: %v", err))
+	}
+	// Wait for a maximum of 5 minutes
+	timeLimit := 300
+	for timeLimit > 0 {
+		time.Sleep(2 * time.Second)
+		output := out.String()
+		if strings.Contains(output, "Processing of DynamoDB Streams started...") {
+			break
+		}
+		timeLimit--
+	}
+	if timeLimit == 0 {
+		return fmt.Errorf("error! streaming migration not started successfully")
+	}
+
+	populateDynamoDBStreams(t)
+
+	// Wait for a maximum of 5 minutes and check row count
+	timeLimit = 300
+	for timeLimit > 0 {
+		time.Sleep(2 * time.Second)
+		// Check count of row to be 2.
+		rowCount, err := rowCountTestTable(client)
```

Comment:
Instead of checking the table counts did we discuss to check for the output msg in logs ?

Reply:
yes, we did discuss that. But the problem was with updating the output every minute. Instead we can just check every 2 minutes whether row count is 2 or not from spanner queries. This will be better in terms of time consumption.